kratos/mpi/sources/mpi_data_communicator.cpp

```diff
@@ -694,6 +709,33 @@ std::vector<TDataType> MPIDataCommunicator::AllReduceDetailVector(
     return reduced_values;
 }
 
+template<class TDataType> 
+std::pair<TDataType, int> MPIDataCommunicator::AllReduceDetailWithLocation(
+    const std::pair<TDataType, int>& rLocalValues,
+    MPI_Op Operation
+    ) const
+{
+    struct {
+        TDataType value;
+        int rank;
+    } local_reduce, global_reduce;
+    local_reduce.value = rLocalValues.first;
+    local_reduce.rank = rLocalValues.second;
+    MPI_Datatype data_type;
+    if constexpr (std::is_same_v<TDataType, double>) {
+        data_type = MPI_DOUBLE_INT;
+    } else if constexpr (std::is_same_v<TDataType, long int>) {
+        data_type = MPI_LONG_INT;
+    } else if constexpr (std::is_same_v<TDataType, int>) {
+        data_type = MPI_2INT;
```

Comment:
Okay

Reply:
Done, I still needed the constexpr because there are many templated specializations not defined for MPIMessage and othwerwise is not possible to even compile